stable/prometheus-operator/templates/alertmanager/secret.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.alertmanager.enabled }}
+{{- if and (.Values.alertmanager.enabled) (eq .Values.alertmanager.alertmanagerConfigExternal false) }}
```

Comment:
I don't think it's a good option. Maybe it's better to change to something like "alertmanager.alertmanagerSpec.defaultConfig" so you won't need to check something for being false?